components/zcash_address/src/kind/unified.rs

```diff
@@ -88,6 +88,12 @@ impl From<Typecode> for u32 {
     }
 }
 
+impl From<Typecode> for usize {
+    fn from(t: Typecode) -> Self {
+        u32::from(t).try_into().unwrap()
```

Comment:
In `pasta_curves` we statically assert that `usize` is at least 32 bits: https://github.com/zcash/pasta_curves/blob/682a0e65b87a835df9cd1e1e55468d969269d992/src/arithmetic/fields.rs#L18 . We should just do the same here.

Reply:
Since this is just for typecodes and is in `zcash_address` (which doesn't provide any operations that require the curves associated with the underlying keys) I think that providing `TryFrom` and allowing the caller to unwrap is fine. In particular, where we're using this it's always safe to unwrap because we're using specific `TypeCode` constants.